olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/Types.h

```diff
@@ -91,6 +92,14 @@ using PrefetchTilesResponseCallback = Callback<PrefetchTilesResult>;
 /// The callback type for the prefetch status update.
 using PrefetchStatusCallback = std::function<void(PrefetchStatus)>;
 
+/// The alias of the prefetch partitions result.
+using PrefetchPartitionsResult =
+    std::vector<std::shared_ptr<PrefetchPartitionResult>>;
+/// The prefetch partitions response type.
+using PrefetchPartitionsResponse = Response<PrefetchPartitionsResult>;
+/// The callback type of the prefetch completion.
+using PrefetchPartitionsResponseCallback = Callback<PrefetchPartitionsResult>;
+
```

Comment:
I would not go the same way as for tile prefetch as it is wrong, as we see now.
I would go with a structure that contains all the list of downloaded partition and not a vector. This way we can extend what information we give back to the user. The `PrefetchPartitionsResponse` will still be Response<> based but it will point to this structure.
We do not need to get the status of each individual tile requested. It should be enough to the user if we give him the successfully downloaded tiles back.